src/backend/web/validation.js

```diff
@@ -50,46 +66,44 @@ function validateNewFeed() {
     }
   };
 }
+// Rules for Validation of query params for the /posts route
+const validatePostsQueryRules = [
+  query('per_page', 'per_page needs to be empty or a integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+  query('page', 'page needs to be empty or an integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+];
 
 // query validation starts here
 // queryValidation rules
-const queryValidationRules = () => {
-  return [
-    // text must be between 1 and 256 and not empty
-    check('text')
-      .exists({ checkFalsy: true })
-      .withMessage('text should not be empty')
-      .bail()
-      .isLength({ max: 256, min: 1 })
-      .withMessage('text should be between 1 to 256 characters')
-      .bail(),
-    // filter must exist and have a valid value
-    check('filter')
-      .exists({ checkFalsy: true })
-      .withMessage('filter should exist')
-      .bail()
-      .isIn(['post', 'author'])
-      .withMessage('invalid filter value')
-      .bail(),
-  ];
-};
-
-// queryValidationFunction
-const validateQuery = () => {
-  return async (req, res, next) => {
-    await Promise.all(queryValidationRules().map((rule) => rule.run(req)));
 
-    const errors = validationResult(req);
-    if (errors.isEmpty()) {
-      return next();
-    }
-    return res.status(400).json({
-      errors: errors.array(),
-    });
-  };
-};
+const queryValidationRules = [
+  // text must be between 1 and 256 and not empty
```

Comment:
@zg3d : I feel I am asking too many things from you. But can you make one last change to change this to postsQueryValidationRules to make it consistent with the queryValidationRules naming from @chawlapalak ? Greatly appreciate your effort.

Reply:
@zg3d : You might have to do another rebase now we found a bug that express-validator was put inside dev-dependencies as supposedly in dependencies. You can see the conversation in Telescope channel